apps/c/pthread_sleep/main.c

```diff
@@ -0,0 +1,61 @@
+#include <pthread.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <time.h>
+#include <unistd.h>
+
+const int NUM_TASKS = 5;
+
+void *tickfunc(void *arg)
+{
+    for (int i = 0; i < 30; i++) {
+        printf("  tick %d\n", i);
+        usleep(500000);
+    }
+    return NULL;
+}
+
+void *tickfunc2(void *arg)
+{
+    pid_t task_id = getpid();
+
+    for (int j = 0; j < 3; j++) {
+        int sec = *(int *)arg + 1;
+        printf("task %d sleep %d seconds (%d) ...\n", task_id, sec, j);
+        time_t now, later;
+        time(&now);
+        sleep(sec);
+        time(&later);
+        printf("task %d actually sleep %d seconds (%d) ...\n", task_id, later - now, j);
+    }
+    return NULL;
+}
+
+void main()
+{
+    printf("Hello, main task!\n");
+    time_t now, later;
+    time(&now);
+    sleep(1);
+    time(&later);
+    printf("main task sleep for %ds\n", later - now);
+
+    pthread_t t1;
+    pthread_create(&t1, NULL, tickfunc, NULL);
+
+    pthread_t tasks[NUM_TASKS] = {0};
```

Comment:
Sorry, it should be `tasks[NUM_TASK + 1]`.
And all below hardcoded number should be `NUM_TASK` or `NUM_TASK + 1`.